src/vte.c

```diff
@@ -675,6 +675,45 @@ const gchar *vte_get_working_directory(void)
 				g_free(cwd);
 			}
 		}
+		else
+		{
+			gint status = 0;
+			gchar *stdout = NULL;
+			gchar *pid_str = g_strdup_printf("%d", pid);
+			gchar *argv[] = {"lsof", "-a", "-d", "cwd", "-F", "n", "-p", pid_str, NULL};
+
+			if (utils_spawn_sync(NULL, argv, NULL, G_SPAWN_SEARCH_PATH,
+					NULL, NULL, &stdout, NULL, &status, NULL))
+			{
+				if (status == 0)
```

Comment:
BTW, I guess the real test her needs to be something like `WIFEXITED(status) && WEXITSTATUS(status == 0)`